packages/eslint-plugin/src/rules/prefer-readonly.ts

```diff
@@ -16,12 +16,10 @@ type Options = [
 ];
 
 const functionScopeBoundaries = [
-  'ArrowFunctionExpression',
-  'FunctionDeclaration',
-  'FunctionExpression',
-  'GetAccessor',
-  'MethodDefinition',
-  'SetAccessor',
```

Comment:
`GetAccessor` and `SetAccessor` is not valid node type, we are converting them in ts-estree to `FunctionExpression`